File: app/models/company.py
# ✅ app/models/company.py

from sqlalchemy import Column, Integer, String, Float
from sqlalchemy.orm import relationship
from app.database.db import Base

class Company(Base):
    __tablename__ = "companies"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, unique=True, index=True)
    city = Column(String)
    website = Column(String)
    linkedin = Column(String)
    email = Column(String)
    vacancies = relationship("Vacancy", back_populates="company", cascade="all, delete")

File: frontend/src/types/Company.ts
export interface Company {
    id: number;
    name: string;
    city: string;
    website: string;
    email: string;
    linkedin: string;
    latitude: number;
    longitude: number;
  }

export type CompanyFormData = Omit<Company, 'id' | 'latitude' | 'longitude'> & {
  latitude: string;
  longitude: string;
};